pkg/controllers/updaterun/controller.go

```diff
@@ -235,28 +235,39 @@ func (r *Reconciler) SetupWithManager(mgr runtime.Manager) error {
 			// We only care about when an approval request is approved.
 			UpdateFunc: func(ctx context.Context, e event.UpdateEvent, q workqueue.RateLimitingInterface) {
 				klog.V(2).InfoS("Handling a clusterApprovalRequest update event", "clusterApprovalRequest", klog.KObj(e.ObjectNew))
-				handleClusterApprovalRequest(e.ObjectNew, q)
-			},
-			GenericFunc: func(ctx context.Context, e event.GenericEvent, q workqueue.RateLimitingInterface) {
-				klog.V(2).InfoS("Handling a clusterApprovalRequest generic event", "clusterApprovalRequest", klog.KObj(e.Object))
-				handleClusterApprovalRequest(e.Object, q)
+				handleClusterApprovalRequest(e.ObjectOld, e.ObjectNew, q)
```

Comment:
just curious, I wonder why removing the genricFunc? I put it there just for safety as I am not 100% sure what is considered "generic".

Reply:
genericFunc handles genericEvents, whose operation is unknown, e.g. triggered by a timer or events outside of the cluster: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/event#GenericEvent. Since I only want to trigger reconcile upon approval status change, I removed the genericFunc.

Reply:
I guess my question is "is it guaranteed that the status (and in other cases, the spec) of an object is not changed when the operation is unknown"?  I haven't spent enough time in the code base to figure that out.